daemon/api.go

```diff
@@ -1604,31 +1672,28 @@ func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response
 		return BadRequest("cannot create user: 'email' field is empty")
 	}
 
-	v, err := storeUserInfo(createData.Email)
-	if err != nil {
-		return BadRequest("cannot create user %q: %s", createData.Email, err)
+	// FIXME: check --known flag and use "getUserDetailsFromAssertions"
+	//        in this case and getUserDetailsFromStore otherwise
+	username, opts, err := getUserDetailsFromAssertion(c.d.overlord.State(), createData.Email)
+	if username == "" {
```

Comment:
Feels like we have a few of entangled issues here, which should either be solved on this branch or on the next one introducing --known. The call above needs to check its error rather than ignoring it altogether when username is empty. Then, an empty username is a bug on the system-user assertion, and we need to flag it as such. Right now the function works all the way through its end as if it was returning something valid, but the username is a required field on this assertion, and it's a bug to not have it which should be reported as such instead of changing behavior in unexpected ways.